dace/config_schema.yml

```diff
@@ -770,6 +770,184 @@ required:
                 title: Print FPGA runtime 
                 description: Prints the runtime of instrumented FPGA kernel states to standard output. 
 
+    #############################################
+    # SOAP I/O analysis settings
+
+    soap:
+        type: dict
+        title: SOAP
+        description: SOAP configuration parameters
+        required:
+            analysis:
+                type: dict
+                title: Analysis
+                description: SOAP analysis parameters
+                required:         
+                    io_analysis:
+                        type: bool
+                        title: Perform I/O analysis
+                        default: true
+                        description: >
+                            Specifies whether the I/O analysis on the SDFG shall be performed
+
+                    wd_analysis:
+                        type: bool
+                        title: Perform work/depth analysis
+                        default: false
+                        description: >
+                            Specifies whether the work-depth analysis on the SDFG shall be performed
+
+                    all_injective:
+                        type: bool
+                        title: Projection of non-SOAP accesses
+                        default: true 
+                        description: >
+                            If access function vectors are not SOAP (e.g., A[i,k] and A[k,j]), should it
+                            be assumed that they never overlap and analyze them as separate logical arrays.
+                    
+            decomposition:
+                type: dict
+                title: Decomposition
+                description: Parallel decomposition and schedule generation
+                required:
+                    chosen_par_setup:
+                        type: int
+                        title: Choose between memory-dependent and memory-independent prarallel schedule
+                        default: 0
+                        description: >
+                            0: memory-dependent
+                            1: memory-independent
+
+                    decomposition_params:
+                        type: list
+                        title: Numerical values of input parameters (list of tuples)
+                        description: >
+                            Should SOAP generate distribute data decomposition, numerical values
+                            of input parameters are required: number of processors P, size of fast memory Ss,
+                            and dimension sizes. E.g.,: [("p", 8), ("Ss", 32*1024), ("S0", 512), ("S1", 512), ("S2", 512)]
+
+            solver:
+                type: dict
+                title: Solver
+                description: Symbolic solver parameters
+                required:
+                    db_path:
```

Comment:
This should have a default value and when you run the solver it should auto-generate the file if it doesn't exist.